analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -57,9 +57,9 @@ protected SonarAnalysisContextBase(SonarAnalysisContext analysisContext, TContex
     /// <param name="tree">Tree to decide on. Can be null for Symbol-based and Compilation-based scenarios. And we want to analyze those too.</param>
     /// <param name="generatedCodeRecognizer">When set, generated trees are analyzed only when language-specific 'analyzeGeneratedCode' configuration property is also set.</param>
     public bool ShouldAnalyzeTree(SyntaxTree tree, GeneratedCodeRecognizer generatedCodeRecognizer) =>
-        SonarLintFile() is var sonarLintReader
-        && (generatedCodeRecognizer is null || sonarLintReader.AnalyzeGeneratedCode || !tree.IsGenerated(generatedCodeRecognizer, Compilation))
-        && (tree is null || (!IsUnchanged(tree) && ShouldAnalyzeFile(sonarLintReader, tree.FilePath)));
+        ReadSonarLintXml() is var sonarLint
+        && (generatedCodeRecognizer is null || sonarLint.AnalyzeGeneratedCode || !tree.IsGenerated(generatedCodeRecognizer, Compilation))
+        && (tree is null || ShouldAnalyzeFile(sonarLint, tree));
```

Comment:
`ShouldAnalyzeFile` doesn't need the tree. It needs the file.
Same as `ShouldAnalyzeTree` needs a tree.
Narrowing down the scope makes it better and prevents accidental cross-usages
```suggestion
        && (tree is null || ShouldAnalyzeFile(sonarLint, tree.FilePath));
```

Reply:
I want the `ShouldAnalyzeFile` to stay there